### PR TITLE
fix: avoid duplicated alias resolution & add comments to module resolution

### PR DESCRIPTION
**Pull request by ryoppippi**
This pull request addresses an issue where module aliases were being resolved multiple times unnecessarily. The changes ensure that if a module is already resolved, it is returned immediately without further alias resolution attempts. This optimization helps in avoiding redundant operations and improves the efficiency of the module resolution process.

### Changes:
- Changed the variable `module` from `let` to `const` to ensure it is not reassigned.
- Added a check to return the module immediately if it is already resolved.
- Added comments to clarify the steps in the module resolution process:
  - If the module is resolved, return it.
  - If no matching alias is found, return the module.
  - If a matching alias is found, resolve the alias and return the resolved module.

### Code Diff:
```diff
- let module = ts.resolveModuleName(lit.text, containingFile, options, host, host.getModuleResolutionCache?.());
+ const module = ts.resolveModuleName(lit.text, containingFile, options, host, host.getModuleResolutionCache?.());

+ /* if module is resolved, return it */
+ if (module.resolvedModule != null) {
+     return module;
+ }

+ /* find matching alias */
const alias = findMatchingAlias(lit.text, aliases);

- if (module.resolvedModule == null && alias != null) {
-     module = ts.resolveModuleName(
-         resolve(lit.text.replace(alias.find, alias.replacement)),
-         containingFile,
-         options,
-         host,
-     );
+ /* if no matching alias, return module */
+ if (alias == null) {
+     return module;
}

- return module;
+ /* resolve alias */
+ return ts.resolveModuleName(
+     resolve(lit.text.replace(alias.find, alias.replacement)),
+     containingFile,
+     options,
+     host,
+ );
```

This change ensures that the module resolution process is more efficient by avoiding duplicated alias resolution.

Context: https://github.com/ryoppippi/unplugin-typia/pull/287#discussion_r1720132649

## Summary by CodeRabbit

- **New Features**
	- Improved module resolution logic for enhanced clarity and efficiency.
	- Streamlined handling of module names and aliases, making the functionality more intuitive for users.
  
- **Bug Fixes**
	- Resolved issues related to nested conditions, improving overall code readability and reliability.

### CODE REVIEW

**Comment by ryoppippi** [comment]
@ansaeb, Could you review it?

**Comment by ryoppippi** [comment]
@asnaeb  Could you review it?

**Comment by ryoppippi** [comment]
@asneab If you review this, that would be fine.

**Comment by ryoppippi** [comment]
@asnaeb If you review this, that would be fine.

**Comment by ryoppippi** [comment]
I'm sorry but I modified some coding style.
I need to fix my eslint config(maybe my eslintconfig has a bug so I fixed it).
Anyway, thank you as always @asnaeb

**Comment by asnaeb** [comment]
May I ask you why you prefer loose over strict equality? Example `if (alias == null)` instead of `if (!alias)`

File: packages/unplugin-typia/src/core/typia.ts [hunk 1]
```diff
@@ -118,17 +118,28 @@ async function getProgramAndSource(
...
+					return module;
 				}
-				return module;
+
+				/* resolve alias */
+				return ts.resolveModuleName(
+					resolve(lit.text.replace(alias.find, alias.replacement)),
+					containingFile,
+					options,
+					host,
```
Comment (asnaeb) [inline packages/unplugin-typia/src/core/typia.ts:150]: We should add `host.getModuleResolutionCache?.()` here as well, I missed it previously

**Comment by ryoppippi** [comment]
@asnaeb 
So this is my coding style.
- I always enable [`strict-boolean-expressions`](strict-boolean-expressions) ( I thought configured `allowNullableObject: false`, but it is not configured, sorry)
- using `==null` to detect null/undefined is widely used (https://basarat.gitbook.io/typescript/recap/null-undefined) (https://eslint.org/docs/latest/rules/eqeqeq#allow-null)

Therefore, I prefer `if (alias == null)` over `if (!alias)`!

File: packages/unplugin-typia/src/core/typia.ts [hunk 2]
```diff
@@ -118,17 +118,28 @@ async function getProgramAndSource(
 	if (aliases != null && aliases.length > 0) {
 		host.resolveModuleNameLiterals = (moduleLiterals, containingFile, _, options) => {
 			return moduleLiterals.map((lit) => {
-				let module = ts.resolveModuleName(lit.text, containingFile, options, host, host.getModuleResolutionCache?.());
+				const module = ts.resolveModuleName(lit.text, containingFile, options, host, host.getModuleResolutionCache?.());
+
+				/* if module is resolved, return it */
+				if (module.resolvedModule != null) {
```
Comment (asnaeb) [inline packages/unplugin-typia/src/core/typia.ts:133]: I would prefer`if (module.resolvedModule)` here. Loose equality makes the code less readable and is not considered good practice but you decide :)

File: packages/unplugin-typia/src/core/typia.ts [hunk 3]
```diff
@@ -118,17 +118,28 @@ async function getProgramAndSource(
...
-				if (module.resolvedModule == null && alias != null) {
-					module = ts.resolveModuleName(
-						resolve(lit.text.replace(alias.find, alias.replacement)),
-						containingFile,
-						options,
-						host,
-					);
+
+				/* if no matching alias, return module */
+				if (alias == null) {
```
Comment (asnaeb) [inline packages/unplugin-typia/src/core/typia.ts:141]: Here as well, I suggest `if (alias)`

File: packages/unplugin-typia/src/core/typia.ts (hunk 2 above)
Comment (ryoppippi) [reply packages/unplugin-typia/src/core/typia.ts:133]: Thank you!

File: packages/unplugin-typia/src/core/typia.ts (hunk 1 above)
Comment (ryoppippi) [reply packages/unplugin-typia/src/core/typia.ts:150]: OK I'll add it!

Comment (ryoppippi) [reply packages/unplugin-typia/src/core/typia.ts:150]: I added at [5e4ea19](https://github.com/ryoppippi/unplugin-typia/pull/288/commits/5e4ea1986f901b9b23537e88db7f8acb894f0903)

**Comment by asnaeb** [comment]
> @asnaeb So this is my coding style.
> 
> * I always enable [`strict-boolean-expressions`](strict-boolean-expressions) ( I thought configured `allowNullableObject: false`, but it is not configured, sorry), because I got so many bugs in my entire life (not only JavaScrcipt but aloso python or other languages)
> * using `==null` to detect null/undefined is widely used (https://basarat.gitbook.io/typescript/recap/null-undefined) (https://eslint.org/docs/latest/rules/eqeqeq#allow-null). I use loose equality only for `==null`.
> 
> Therefore, I prefer `if (alias == null)` over `if (!alias)`!

Okay, nice, I think we should be good to go now 👍🏻

**Comment by ryoppippi** [comment]
Nice!
Thank you for taking your time to review it!
I'll merge it and release it

**Comment by ryoppippi** [comment]
@asnaeb 
I just released a new version!
Thank you for today!
Good night!

**Comment by asnaeb** [comment]
> @asnaeb I just released a new version! Thank you for today! Good night!

No problem, I'm looking forward to contribute more

**Pull request merged**